libvcx/src/messages/get_message.rs

```diff
@@ -539,55 +540,56 @@ mod tests {
-                                                                             "credential_name".to_string(),
-                                                                             credential_data.to_owned(),
-                                                                             1).unwrap();
+        let (alice_to_faber, faber_to_alice) = ::connection::tests::create_connected_connections();
 
-        ::issuer_credential::send_credential_offer(credential_offer, alice).unwrap();
+        info!("test_download_messages :: going send message from faber to alice");
+        send_generic_message(faber_to_alice, "Hello Alice", &json!({"msg_type": "toalice", "msg_title": "msg1"}).to_string()).unwrap();
+        info!("test_download_messages :: messge sent");
+        send_generic_message(faber_to_alice, "How are you Alice?", &json!({"msg_type": "toalice", "msg_title": "msg2"}).to_string()).unwrap();
 
         thread::sleep(Duration::from_millis(1000));
 
-        let hello_uid = ::connection::send_generic_message(alice, "hello", &json!({"msg_type":"hello", "msg_title": "hello", "ref_msg_id": null}).to_string()).unwrap();
-
         // AS CONSUMER GET MESSAGES
         ::utils::devsetup::set_consumer();
-
-        let _all_messages = download_messages(None, None, None).unwrap();
-
-        let pending = download_messages(None, Some(vec!["MS-103".to_string()]), None).unwrap();
-        assert_eq!(pending.len(), 1);
-        assert!(pending[0].msgs[0].decrypted_payload.is_some());
-
-        let accepted = download_messages(None, Some(vec!["MS-104".to_string()]), None).unwrap();
-        assert_eq!(accepted[0].msgs.len(), 2);
-
-        let specific = download_messages(None, None, Some(vec![accepted[0].msgs[0].uid.clone()])).unwrap();
+        send_generic_message(alice_to_faber, "Hello Faber", &json!({"msg_type": "tofaber", "msg_title": "msg1"}).to_string()).unwrap();
+
+        info!("test_download_messages :: going to download all messages");
+        let all_messages = download_messages(None, None, None).unwrap();
+        info!("_all_messages = {}", serde_json::to_string(&all_messages).unwrap());
+        assert_eq!(all_messages.len(), 1);
+        assert_eq!(all_messages[0].msgs.len(), 3);
+        assert!(all_messages[0].msgs[0].decrypted_payload.is_some());
+        assert!(all_messages[0].msgs[1].decrypted_payload.is_some());
+
+        let received = download_messages(None, Some(vec![MessageStatusCode::Received.to_string()]), None).unwrap();
+        assert_eq!(received.len(), 1);
+        assert_eq!(received[0].msgs.len(), 2);
+        assert!(received[0].msgs[0].decrypted_payload.is_some());
+        assert_eq!(received[0].msgs[0].status_code, MessageStatusCode::Received);
+        assert!(received[0].msgs[1].decrypted_payload.is_some());
+
+        // there should be review aries message connections/1.0/response from Aries-Faber connection protocol
```

Comment:
Mark this as TODO

Reply:
this is misunderstanding, I've clarified the comment